File: Src/Icm.Core.Tests/Basic types extensions/TimespanExtensionsTest.vb
Imports System


<TestFixture(), Category("Icm")>
Public Class UndefinedValueExtensionsTest


End Class


'''<summary>
'''This is a test class for TimespanExtensionsTest and is intended
'''to contain all TimespanExtensionsTest Unit Tests
'''</summary>
<TestFixture(), Category("Icm")>
Public Class TimespanExtensionsTest

    '''<summary>
    '''A test for DividedBy
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub DividedByTest()
        Dim t As TimeSpan = New TimeSpan(3, 0, 0)
        Dim divisor As Single = 3
        Dim expected As TimeSpan = New TimeSpan(CLng(t.Ticks / 3))
        Dim actual As TimeSpan

        'bassic case
        actual = TimespanExtensions.DividedBy(t, divisor)
        Assert.AreEqual(expected, actual)

        'Divisor = 0
        divisor = 0
        Try
            actual = TimespanExtensions.DividedBy(t, divisor)
            Assert.Fail("Should throw OverFlowException")
        Catch ex As OverflowException

        Catch ex As Exception
            Assert.Fail("Should throw OverFlowException")
        End Try

    End Sub

    '''<summary>
    '''A test for IsZero
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub IsZeroTest()

        Dim t As TimeSpan = New TimeSpan(3, 0, 0)
        Dim expected As Boolean = False
        Dim actual As Boolean
        actual = TimespanExtensions.IsZero(t)
        Assert.AreEqual(expected, actual)

        'Caso 2 (la fecha es igual a 0)
        Dim d As TimeSpan = New TimeSpan(0, 0, 0)
        expected = True
        actual = TimespanExtensions.IsZero(d)
        Assert.AreEqual(expected, actual)
    End Sub

    '''<summary>
    '''A test for IsZero
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub IsNotZerotest()

        Dim t As TimeSpan = New TimeSpan(3, 0, 0)
        Dim expected As Boolean = False
        Dim actual As Boolean
        actual = TimespanExtensions.IsZero(t)
        Assert.AreEqual(expected, actual)

    End Sub


    '''<summary>
    '''A test for ToAbrev
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub ToAbrevTest()
        Dim ts As TimeSpan = New TimeSpan()
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        ts = New TimeSpan(2, 14, 2)
        expected = "2h14'2''"
        actual = TimespanExtensions.ToAbbrev(ts)
        Assert.AreEqual(expected, actual)

        'Caso 1
        ts = New TimeSpan(0, 0, 2)
        expected = "2''"
        actual = TimespanExtensions.ToAbbrev(ts)
        Assert.AreEqual(expected, actual)

        'Caso 3
        ts = New TimeSpan(-5, 14, 18)
        expected = "-4h-45'-42''"
        actual = TimespanExtensions.ToAbbrev(ts)
        Assert.AreEqual(expected, actual)

    End Sub

    '''<summary>
    '''A test for ToHHmm
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub ToHHmmTest()
        Dim ts As TimeSpan = New TimeSpan()
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        ts = New TimeSpan(2, 14, 2)
        expected = "02:14"
        actual = TimespanExtensions.ToHHmm(ts)
        Assert.AreEqual(expected, actual)

        'Caso 2
        ts = New TimeSpan(-5, 14, 18)
        expected = "-04:45"
        actual = TimespanExtensions.ToHHmm(ts)
        Assert.AreEqual(expected, actual)

    End Sub

    '''<summary>
    '''A test for ToHHmmss
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub ToHHmmssTest()
        Dim ts As TimeSpan
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        ts = New TimeSpan(2, 14, 18)
        expected = "02:14:18"
        actual = TimespanExtensions.ToHHmmss(ts)
        Assert.AreEqual(expected, actual)

        'Caso 2
        ts = New TimeSpan(-5, 14, 18)
        expected = "-04:45:42"
        actual = TimespanExtensions.ToHHmmss(ts)
        Assert.AreEqual(expected, actual)

        'Caso 3
        ts = New TimeSpan(0, 0, 0)
        expected = "00:00:00"
        actual = TimespanExtensions.ToHHmmss(ts)
        Assert.AreEqual(expected, actual)

    End Sub

    '''<summary>
    '''A test for ToMicroseconds
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub ToMicrosecondsTest()
        Dim ts As TimeSpan = New TimeSpan()
        Dim expected As Long = 0
        Dim actual As Long

        'Caso 1
        ts = New TimeSpan(2, 14, 18, 25)
        expected = 224305000000
        actual = TimespanExtensions.ToMicroseconds(ts)
        Assert.AreEqual(expected, actual)

        'Caso 2
        ts = New TimeSpan(0, 0, 0, 2)
        expected = 2000000
        actual = TimespanExtensions.ToMicroseconds(ts)
        Assert.AreEqual(expected, actual)

        'Caso 2
        ts = New TimeSpan(-5, 14, 18)
        expected = -17142000000
        actual = TimespanExtensions.ToMicroseconds(ts)
        Assert.AreEqual(expected, actual)

    End Sub

    '''<summary>
    '''A test for ToMillisecondsAndOne
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub ToMillisecondsAndOneTest()
        Dim ts As TimeSpan = New TimeSpan()
        Dim expected As String = String.Empty
        Dim actual As String

        'Caso 1
        ts = New TimeSpan(2, 14, 18, 25)
        expected = "224305000,0"
        actual = TimespanExtensions.ToMillisecondsAndOne(ts)
        Assert.AreEqual(expected, actual)

        'Caso 2
        ts = New TimeSpan(0, 0, 0, 2)
        expected = "2000,0"
        actual = TimespanExtensions.ToMillisecondsAndOne(ts)
        Assert.AreEqual(expected, actual)

        'Caso 3
        ts = New TimeSpan(-5, 14, 18)
        expected = "-17142000,0"
        actual = TimespanExtensions.ToMillisecondsAndOne(ts)
        Assert.AreEqual(expected, actual)
    End Sub

    '''<summary>
    '''A test for Tommssttt
    '''</summary>
    <Test(), Category("Icm")>
    Public Sub TommsstttTest()
        Dim ts As TimeSpan = New TimeSpan()
        Dim expected As String = String.Empty
        Dim actual As String


        'Caso 1
        ts = New TimeSpan(2, 14, 18, 25)
        expected = "3738:25.000"
        actual = TimespanExtensions.Tommssttt(ts)
        Assert.AreEqual(expected, actual)

        'Caso 2
        ts = New TimeSpan(0, 0, 0, 2)
        expected = "00:02.000"
        actual = TimespanExtensions.Tommssttt(ts)
        Assert.AreEqual(expected, actual)

        'Caso 3
        ts = New TimeSpan(-5, 14, 18)
        expected = "-285:42.000"
        actual = TimespanExtensions.Tommssttt(ts)
        Assert.AreEqual(expected, actual)
    End Sub
End Class

File: Src/Icm.Core.Tests/Basic types extensions/TimespanExtensionsTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm;
using NUnit.Framework;

[TestFixture(), Category("Icm")]
public class TimespanExtensionsTest
{

	static readonly object[] DividedByTestCases = {
		new TestCaseData(TimeSpan.FromHours(3), 3).Returns(TimeSpan.FromHours(1)),
		new TestCaseData(TimeSpan.FromHours(-3), 3).Returns(TimeSpan.FromHours(-1)),
		new TestCaseData(TimeSpan.FromHours(3), -3).Returns(TimeSpan.FromHours(-1)),
		new TestCaseData(TimeSpan.Zero, 3).Returns(TimeSpan.Zero),
		new TestCaseData(TimeSpan.FromHours(3), 0).Throws(typeof(OverflowException))

	};
	[TestCaseSource(nameof(DividedByTestCases))]
	public TimeSpan DividedBy_Test(TimeSpan span, float divisor)
	{
		return span.DividedBy(divisor);
	}

	static readonly object[] IsZeroTestCases = {
		new TestCaseData(TimeSpan.FromHours(3)).Returns(false),
		new TestCaseData(TimeSpan.FromHours(-3)).Returns(false),
		new TestCaseData(TimeSpan.Zero).Returns(true)

	};
	[TestCaseSource(nameof(IsZeroTestCases))]
	public bool IsZero_Test(TimeSpan target)
	{
		return target.IsZero();
	}

	static readonly object[] IsNotZeroTestCases = {
		new TestCaseData(TimeSpan.FromHours(3)).Returns(true),
		new TestCaseData(TimeSpan.FromHours(-3)).Returns(true),
		new TestCaseData(TimeSpan.Zero).Returns(false)

	};
	[TestCaseSource(nameof(IsNotZeroTestCases))]
	public bool IsNotZero_Test(TimeSpan target)
	{
		return target.IsNotZero();
	}

	static readonly object[] ToAbbrevTestCases = {
		new TestCaseData(new TimeSpan(2, 14, 2)).Returns("2h14'2''"),
		new TestCaseData(new TimeSpan(0, 0, 2)).Returns("2''"),
		new TestCaseData(TimeSpan.Zero).Returns("0"),
		new TestCaseData(new TimeSpan(-5, 14, 18)).Returns("-4h45'42''")

	};
	[TestCaseSource(nameof(ToAbbrevTestCases))]
	public string ToAbbrev_Test(TimeSpan target)
	{
		return target.ToAbbrev;
	}

	static readonly object[] ToHHmmTestCases = {
		new TestCaseData(new TimeSpan(2, 14, 18)).Returns("02:14"),
		new TestCaseData(new TimeSpan(0, 0, 2)).Returns("00:00"),
		new TestCaseData(TimeSpan.Zero).Returns("00:00"),
		new TestCaseData(new TimeSpan(-5, 14, 18)).Returns("-04:45")

	};
	[TestCaseSource(nameof(ToHHmmTestCases))]
	public string ToHHmm_Test(TimeSpan target)
	{
		return target.ToHHmm;
	}

	static readonly object[] ToHHmmssTestCases = {
		new TestCaseData(new TimeSpan(2, 14, 18)).Returns("02:14:18"),
		new TestCaseData(new TimeSpan(0, 0, 2)).Returns("00:00:02"),
		new TestCaseData(TimeSpan.Zero).Returns("00:00:00"),
		new TestCaseData(new TimeSpan(-5, 14, 18)).Returns("-04:45:42")

	};
	[TestCaseSource(nameof(ToHHmmssTestCases))]
	public string ToHHmmss_Test(TimeSpan target)
	{
		return target.ToHHmmss;
	}

	static readonly object[] TommsstttTestCases = {
		new TestCaseData(new TimeSpan(2, 14, 18, 25)).Returns("3738:25.000"),
		new TestCaseData(new TimeSpan(0, 0, 2)).Returns("00:02.000"),
		new TestCaseData(TimeSpan.Zero).Returns("00:00.000"),
		new TestCaseData(new TimeSpan(-5, 14, 18)).Returns("-285:42.000")

	};
	[TestCaseSource(nameof(TommsstttTestCases))]
	public string Tommssttt_Test(TimeSpan target)
	{
		return target.Tommssttt;
	}

	static readonly object[] ToHHmmsstttTestCases = {
		new TestCaseData(new TimeSpan(2, 14, 18, 25)).Returns("62:18:25.000"),
		new TestCaseData(new TimeSpan(0, 0, 2)).Returns("00:00:02.000"),
		new TestCaseData(TimeSpan.Zero).Returns("00:00:00.000"),
		new TestCaseData(new TimeSpan(-5, 14, 18)).Returns("-04:45:42.000")

	};
	[TestCaseSource(nameof(ToHHmmsstttTestCases))]
	public string ToHHmmssttt_Test(TimeSpan target)
	{
		return target.ToHHmmssttt;
	}

	static readonly object[] TotalMicrosecondsTestCases = {
		new TestCaseData(new TimeSpan(2, 14, 18, 25)).Returns(224305000000L),
		new TestCaseData(new TimeSpan(0, 0, 0, 2)).Returns(2000000),
		new TestCaseData(TimeSpan.Zero).Returns(0),
		new TestCaseData(new TimeSpan(-5, 14, 18)).Returns(-17142000000L)

	};
	[TestCaseSource(nameof(TotalMicrosecondsTestCases))]
	public long TotalMicroseconds_Test(TimeSpan target)
	{
		return target.TotalMicroseconds;
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
